src/api/tenderly/tenderlyApi.ts

```diff
@@ -0,0 +1,161 @@
+  }
+}
+
+function _get(networkId: Network, url: string): Promise<Response> {
+  const baseUrl = _getApiBaseUrl(networkId)
+  return fetch(baseUrl + url)
+}
+
+function _fetchTrace(networkId: Network, txHash: string): Promise<Trace> {
+  const queryString = `/trace/${txHash}`
+  console.log(`[tenderlyApi:fetchTrace] Fetching trace tx ${txHash} on network ${networkId}`)
+
+  return fetchQuery<Trace>({ get: () => _get(networkId, queryString) }, queryString)
+}
+
+function _fetchTradesAccounts(networkId: Network, txHash: string): Promise<Contract[]> {
+  const queryString = `/tx/${txHash}/contracts`
+  console.log(`[tenderlyApi:fetchTradesAccounts] Fetching tx trades account on network ${networkId}`)
+
+  return fetchQuery<Array<Contract>>({ get: () => _get(networkId, queryString) }, queryString)
+}
+
+export async function getTradesAndTransfers(networkId: Network, txHash: string): Promise<TxTradesAndTransfers> {
+  const trace = await _fetchTrace(networkId, txHash)
+
+  return traceToTransfersTrades(trace)
+}
+
+export function traceToTransfersTrades(trace: Trace): TxTradesAndTransfers {
+  const transfers: Array<Transfer> = []
+  const trades: Array<Trade> = []
+
+  try {
+    trace.logs.forEach((log) => {
+      if (log.name === TypeOfTrace.TRANSFER) {
+        transfers.push({
+          token: log.raw.address,
+          from: log.inputs[IndexTransferInput.from].value,
+          to: log.inputs[IndexTransferInput.to].value,
+          value: log.inputs[IndexTransferInput.value].value,
+        })
+      } else if (log.name === TypeOfTrace.TRADE) {
+        const trade = {
+          owner: log.inputs[IndexTradeInput.owner].value,
+          sellToken: log.inputs[IndexTradeInput.sellToken].value,
+          buyToken: log.inputs[IndexTradeInput.buyToken].value,
+          sellAmount: log.inputs[IndexTradeInput.sellAmount].value,
+          buyAmount: log.inputs[IndexTradeInput.buyAmount].value,
+          feeAmount: log.inputs[IndexTradeInput.feeAmount].value,
+          orderUid: log.inputs[IndexTradeInput.orderUid].value,
+        }
+        if (trade.buyToken === ETH_NULL_ADDRESS) {
+          //ETH transfers are not captured by ERC20 events, so we need to manyally add them to the Transfer list
+          transfers.push({
+            token: ETH_NULL_ADDRESS,
+            from: log.raw.address,
+            to: trade.owner,
+            value: trade.buyAmount,
+          })
+        }
+        trades.push(trade)
+      }
+    })
+  } catch (error) {
+    console.error(`Unable to analyze the JSON trace trades`, error)
+    throw new Error(`Failed to parse the JSON of tenderly trace API`)
+  }
+
+  return { transfers, trades }
+}
+
+export async function getTradesAccount(
+  networkId: Network,
+  txHash: string,
+  trades: Array<Trade>,
+  transfers: Array<Transfer>,
+): Promise<Map<string, Account>> {
+  const contracts = await _fetchTradesAccounts(networkId, txHash)
+
+  return contractsInvolved(contracts, trades, transfers)
```

Comment:
I'm a bit confused that the function tells me it's getting one account for the trades, but the return is a call to `contractsInvolved` function